File: Classes/WMISet/WMISet/Examples/delphi/WmiMethods/Service/FrmNewHostU.pas
unit FrmNewHostU;

interface
{$I define.inc}

uses
  Windows, Messages, SysUtils, Classes, Graphics, Controls, Forms,
  Dialogs,
  {$IFDEF Delphi6}
  Variants,
  {$ENDIF}
  StdCtrls;

type
  TFrmNewHost = class(TForm)
    Label1: TLabel;
    edtHostName: TEdit;
    grbCredentials: TGroupBox;
    lblDomain: TLabel;
    edtDomain: TEdit;
    edtUserName: TEdit;
    lblUserName: TLabel;
    lblPassword: TLabel;
    edtPassword: TEdit;
    btnConnect: TButton;
    btnCancel: TButton;
    procedure EditChanged(Sender: TObject);
  private
    function GetMachineName: widestring;
    procedure SetMachineName(const Value: widestring);
    function GetUserName: widestring;
    { Private declarations }
  public
    { Public declarations }
    property MachineName: widestring read GetMachineName write SetMachineName;
    property UserName: widestring read GetUserName;
  end;

var
  edtUserName: TFrmNewHost;

implementation

{$R *.dfm}

procedure TFrmNewHost.EditChanged(Sender: TObject);
begin
  btnConnect.Enabled := (Trim(edtHostName.Text) <> '') and
                       (Trim(edtUserName.Text) <> '');
end;

function TFrmNewHost.GetMachineName: widestring;
begin
  Result := edtHostName.Text; 
end;

function TFrmNewHost.GetUserName: widestring;
begin
  Result := edtUserName.Text;
  if Trim(edtDomain.Text) <> '' then
    Result := edtDomain.Text + '\' + Result;
end;

procedure TFrmNewHost.SetMachineName(const Value: widestring);
begin
  edtHostName.Text := Value;
  edtHostName.Enabled := Value = '';
end;

end.

File: Classes/WMISet/WMISet/Examples/cbuilder/DiskQuotas/FrmNewHostU.cpp
//---------------------------------------------------------------------------

#include <vcl.h>
#pragma hdrstop

#include "FrmNewHostU.h"
//---------------------------------------------------------------------------
#pragma package(smart_init)
#pragma resource "*.dfm"
TFrmNewHost *FrmNewHost;
//---------------------------------------------------------------------------
__fastcall TFrmNewHost::TFrmNewHost(TComponent* Owner)
        : TForm(Owner) {
};

//---------------------------------------------------------------------------
void __fastcall TFrmNewHost::SetMachineName(WideString Value) {
  edtHostName->Text = Value;
  edtHostName->Enabled = (Value.Length() == 0);
}

//---------------------------------------------------------------------------
WideString __fastcall TFrmNewHost::GetUserName() {
  WideString result = edtUserName->Text;
  if (edtDomain->Text.Trim() != "") {
        result = edtDomain->Text + "\\" + result;
  }
  return result;
}

//---------------------------------------------------------------------------
WideString __fastcall TFrmNewHost::GetMachineName() {
        return edtHostName->Text;
}


void __fastcall TFrmNewHost::OnEditChanged(TObject *Sender)
{
  btnConnect->Enabled = edtHostName->Text.Trim().Length() != 0 &&
                        edtUserName->Text.Trim().Length() != 0;
}
//---------------------------------------------------------------------------

